plugin/path_impersonated_account_secrets.go

```diff
@@ -0,0 +1,87 @@
+package gcpsecrets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
+)
+
+func pathImpersonatedAccountSecretAccessToken(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%s/%s/token", impersonatedAccountPathPrefix, framework.GenericNameRegex("name")),
+		Fields: map[string]*framework.FieldSchema{
+			"name": {
+				Type:        framework.TypeString,
+				Description: "Required. Name of the impersonated account.",
+			},
+		},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation:   &framework.PathOperation{Callback: b.pathImpersonatedAccountAccessToken},
+			logical.UpdateOperation: &framework.PathOperation{Callback: b.pathImpersonatedAccountAccessToken},
+		},
+		HelpSynopsis:    pathTokenHelpSyn,
+		HelpDescription: pathTokenHelpDesc,
+	}
+}
+
+func (b *backend) pathImpersonatedAccountAccessToken(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	acctName := d.Get("name").(string)
+
+	acct, err := b.getImpersonatedAccount(acctName, ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if acct == nil {
+		return logical.ErrorResponse("impersonated account %q does not exists", acctName), nil
+	}
+
+	creds, err := b.credentials(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := getConfig(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	warnings := []string{}
+	acctTtl := time.Duration(acct.Ttl) * time.Second
+	if acctTtl > config.MaxTTL {
+		warnings = append(warnings, fmt.Sprintf("using backend max ttl %q which is less than impersonated account ttl %q for token",
+			config.MaxTTL.String(),
+			acctTtl.String()))
+		acctTtl = config.MaxTTL
+	} else if acctTtl == 0 {
+		warnings = append(warnings, fmt.Sprintf("using backend default ttl %q since impersonated account ttl not configured for token",
+			config.TTL.String()))
+		acctTtl = config.TTL
+	}
+
+	tokenSource, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+		TargetPrincipal: acct.EmailOrId,
+		Scopes:          acct.TokenScopes,
+		Lifetime:        acctTtl,
```

Comment:
Can you confirm that these access tokens aren't different in terms of their lifecycle from [access tokens](https://cloud.google.com/docs/authentication/token-types#at-lifetime) generated via rolesets and static accounts? I mostly want to be sure that they shouldn't be leased secrets for a similar reason explained in [this section](https://developer.hashicorp.com/vault/docs/secrets/gcp#deprecation-of-access-token-leases) of our docs.

From reading the GCP [access token lifetime](https://cloud.google.com/docs/authentication/token-types#access) docs and testing this manually, my guess is they're the same. I'm not aware of a way to revoke these access tokens before their expiration, so it makes sense they're not leased in Vault. I supposed we could eventually support renewal via refresh tokens, but we don't need to add that now. I was able to confirm the tokens are revoked after their expiry and have the 1hr cap by default.

Reply:
> Can you confirm that these access tokens aren't different in terms of their lifecycle from [access tokens](https://cloud.google.com/docs/authentication/token-types#at-lifetime) generated via rolesets and static accounts?

Yes, that is correct. The access tokens returned from this work have the same lifecycle as those access tokens from rolesets and static accounts.

Reply:
Thanks! That makes sense.